apps/nerves_hub_device/lib/nerves_hub_device/ssl.ex

```diff
@@ -162,13 +162,32 @@ defmodule NervesHubDevice.SSL do
     end
   end
 
-  defp check_device_exists(cn, org_id) do
+  defp maybe_jitp_device(cn, %{org_id: org_id, jitp: nil}) do
     case Devices.get_device_by(identifier: cn, org_id: org_id) do
       {:ok, _d} = resp -> resp
       _ -> :device_not_registered
     end
   end
 
+  defp maybe_jitp_device(cn, %{org_id: org_id, jitp: %Devices.CACertificate.JITP{} = jitp}) do
+    case Devices.get_device_by(identifier: cn, org_id: org_id) do
+      {:ok, _d} = resp ->
+        resp
+
+      _ ->
+        case Devices.create_device(%{
+               identifier: cn,
+               org_id: org_id,
+               product_id: jitp.product_id,
+               tags: jitp.tags,
+               description: jitp.description
+             }) do
+          {:ok, _d} = resp -> resp
+          _ -> :device_not_registered
```

Comment:
```suggestion
          _ -> :device_registration_failed
```

It might be nice for more explicit error here